### PR TITLE
don't initialize cuda at import time

### PR DESCRIPTION
**Pull request by technillogue**
# Description

Currently importing `torch_tensorrt` calls `_cuda_init`, which can fail, preventing torch_tensorrt being imported in environments without a (working) GPU. This makes it slow to import torch_tensorrt and isn't necessary. This change switches to determining the current devices when TensorRTCompileSpec is called rather than when it's defined.

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [x] I have made corresponding changes to the documentation
- [x] I have added tests to verify my fix or my feature
- [x] New and existing unit tests pass locally with my changes
- [ ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: py/torch_tensorrt/ts/_compile_spec.py [hunk 1]
```diff
@@ -307,7 +307,7 @@ def _parse_compile_spec(compile_spec_: Dict[str, Any]) -> _ts_C.CompileSpec:
 def TensorRTCompileSpec(
     inputs: Optional[List[torch.Tensor | Input]] = None,
     input_signature: Optional[Any] = None,
-    device: torch.device | Device = Device._current_device(),
+    device: torch.device | Device | None = None,
```
Comment (narendasan) [inline py/torch_tensorrt/ts/_compile_spec.py:310]: Could do `Optional[torch.device | Device]`

**Review by peri044 — COMMENTED** [review]
@technillogue  This looks good. Can you make the device and Optional type as Naren pointed out ? Then, we can merge

**Comment by technillogue** [comment]
I've made the change

**Review by narendasan — APPROVED** [review]
Cool, Looks good to me Thanks!
